### ISSUE TITLE
Working example

### ISSUE THREAD

**Issue by Kajvdh** [bitwalker/exirc#93]
Hey,

I'm trying to get the example on the README.md page working but I'm getting the error:

```
** (Mix) Could not start application examplebot: exited in: FA.start(:normal, [])
    ** (EXIT) an exception was raised:
        ** (ArgumentError) The module ExampleConnectionHandler was given as a child to a supervisor
but it does not implement child_spec/1.
```

I already tried to use `use GenServer` in the `ExampleConnectionHandler` to automatically define this function, this gets rid of this error - but brings me to the next one:

```
** (FunctionClauseError) no function clause matching in GenServer.whereis/1
            (elixir 1.10.4) lib/gen_server.ex:1195: GenServer.whereis([#PID<0.212.0>])
```

Is there any example available that I can use to get started? 

Kind regards,

Kaj

**Comment by tchoutri** [comment]
@Kajvdh Hi, and thank you for this report! Could you please tell me what is the Elixir version you're using? :)
`GenServer.whereis` has been around for at least three years, so I'm a bit worried that you are using an outdated version.

**Comment by tchoutri** [comment]
> Is there any example available that I can use to get started?

To anwser your question: Yes, there are projects using this library listed at the bottom of the README.

**Comment by Kajvdh** [comment]
Hey @tchoutri thanks for the fast response. I'm on the latest elixir version (1.10.4), however I did some further research and found what the issue was.

It was situated in the `start_link` block of the ExampleConnectionHandler that I used from the readme, which was:

```
def start_link(client, state \\ %State{}) do
    GenServer.start_link(__MODULE__, [%{state | client: client}])
end
```

However, the client is passed as a type `List` of the client. So I changed it to:

```
def start_link([client], state \\ %State{}) do
      GenServer.start_link(__MODULE__, [%{state | client: client}])
end
```

Now everything works fine. I'm not sure why I'm the only one having this issue - maybe someone can doublecheck it and correct it in the example if it's indeed always an issue.
